hudi-integ-test/src/main/java/org/apache/hudi/integ/testsuite/HoodieTestSuiteJob.java

```diff
@@ -147,7 +147,7 @@ public void runTestSuite() {
       long startTime = System.currentTimeMillis();
       WriterContext writerContext = new WriterContext(jsc, props, cfg, keyGenerator, sparkSession);
       writerContext.initContext(jsc);
-      DagScheduler dagScheduler = new DagScheduler(workflowDag, writerContext);
+      DagScheduler dagScheduler = new DagScheduler(workflowDag, writerContext, jsc, cfg.numRounds, cfg.delayMins);
```

Comment:
@nsivabalan 
Can we avoid introducing these high level arguments numRound and delayMins to the dagScheduler ? This messes with a simpler design and adds unnecessary overloading to the constructor. 
Instead, we could do something like this

Introduce a top level yaml change around this
dag_name:
dag_rounds:
dag_intermittent_delay:
dag_props:
dag_content:
    insert_node:
        ...
    upsert_node:
        ...
You can then introduce a concept called RuntimeComposableDagNode that basically wraps this concept.

Let me know.